Source/core/ThreadPool.cpp

```diff
@@ -0,0 +1,234 @@
+uint8_t WPEFramework::Core::ThreadPool::CrashMonitor::_numberOfWorkerThread = 0;
+WPEFramework::Core::ThreadPool::CrashMonitor::SigAction WPEFramework::Core::ThreadPool::CrashMonitor::_orig_act={0};
+#endif /* THUNDER_CRASH_HANDLER */
+
+namespace WPEFramework
+{
+    namespace Core
+    {
+#ifdef THUNDER_CRASH_HANDLER
+        ThreadPool::CrashMonitor::CrashMonitor(std::list<Executor>& _units)
+             : _threadlist(_units)
+        {
+            syslog(LOG_NOTICE, "CrashMonitor constructor.");
+        }
+
+        ThreadPool::CrashMonitor::~CrashMonitor()
+        {
+            DeleteCrashMonitor();
+        }
+
+        void ThreadPool::CrashMonitor::SetupCrashHandling()
+        {
+            _adminLock.Lock();
+            SigAction current_act;
+
+            sigaction(SIGSEGV, nullptr, &current_act);
+
+            if (current_act.sa_sigaction != signal_segv)
+            {
+                SigAction sa;
+
+                memset(&sa, 0, sizeof(SigAction));
+                sigemptyset(&sa.sa_mask);
+                sa.sa_sigaction = signal_segv;
+                sa.sa_flags = SA_SIGINFO;
+                _orig_act = current_act;
+                sigaction(SIGSEGV,&sa,nullptr);
+            }
+            _adminLock.Unlock();
+        }
+
+        void ThreadPool::CrashMonitor::CreateCrashMonitor()
+        {
+            if (_threadlist.size())
+            {
+                std::list<WPEFramework::Core::ThreadPool::Executor>::iterator ptr = _threadlist.begin();
+                uint8_t count = 0;
+
+                syslog(LOG_NOTICE, "Updating the thread ids in threadlist array.");
+
+                _numberOfWorkerThread = _threadlist.size();
+
+                _workerThreadInfo = new WPEFramework::Core::ThreadPool::CrashMonitor::WorkerThreadInfo[_numberOfWorkerThread];
+                ASSERT( NULL != _workerThreadInfo);
+
+                for (ptr = _threadlist.begin(), count = 0; ptr!=_threadlist.end(); ptr++, count++)
+                {
+                    _workerThreadInfo[count].threadId = ptr->Id();
+                    _workerThreadInfo[count].jsonRPCString = "";
+                }
+            }
+        }
+
+        void ThreadPool::CrashMonitor::StoreRequestString(::ThreadId threadId, const string& jsonString)
+        {
+            ASSERT( NULL != _workerThreadInfo);
+
+            for (uint8_t  index = 0; index < _numberOfWorkerThread; ++index)
+            {
+                if (threadId == _workerThreadInfo[index].threadId)
+                {
+                  _workerThreadInfo[index].jsonRPCString = jsonString;
+                }
+            }
+        }
+
+        void ThreadPool::CrashMonitor::DeleteCrashMonitor()
+        {
+            if (NULL != _workerThreadInfo)
+            {
+                delete[] _workerThreadInfo;
+                _workerThreadInfo = NULL;
+            }
+        }
+
+        void ThreadPool::CrashMonitor::DumpBacktrace()

```

Comment:
This functionality is already in the THunder core. Do not repeat!! See: https://github.com/rdkcentral/Thunder/blob/R2/Source/core/Portability.cpp#L209-L244